### PR TITLE
EMBCESSMOD-1847, EMBCESSMOD-1848

### PR DESCRIPTION
**Pull request by josephweinkam**
implement EvacuationFileNotesQuery. Update API endpoints to edit note content/isHidden

### CODE REVIEW

File: ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs [hunk 1]
```diff
@@ -310,15 +310,22 @@ public async Task<VerifySecurityPhraseResponse> Handle(VerifySecurityPhraseQuery
 
         public async Task<string> Handle(SaveEvacuationFileNoteCommand cmd)
         {
+            if (string.IsNullOrEmpty(cmd.Note.Id) && string.IsNullOrEmpty(cmd.FileId)) throw new ArgumentNullException("NoteId and FileId cannot both be empty/null.");
```
Comment (ytqsl) [inline ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs:313]: it is better to split the checks to different exceptions (better clarity)

File: ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs [hunk 2]
```diff
@@ -310,15 +310,22 @@ public async Task<VerifySecurityPhraseResponse> Handle(VerifySecurityPhraseQuery
...
+
             var note = mapper.Map<Resources.Cases.Note>(cmd.Note);
             var id = (await caseRepository.ManageCase(new SaveEvacuationFileNote { FileId = cmd.FileId, Note = note })).CaseId;
             return id;
         }
 
-        public async Task<EvacuationFileNotesQueryResult> Handle(EvacuationFileNotesQuery query)
+        public async Task<EvacuationFileNotesQueryResult> Handle(Shared.Contracts.Submissions.EvacuationFileNotesQuery query)
```
Comment (ytqsl) [inline ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs:320]: can we maintain a convention of Response suffix for query return type, and Result suffix for a particular item in a response?

File: ess/src/API/EMBC.ESS/Resources/Cases/CaseRepository.cs [hunk 3]
```diff
@@ -55,6 +55,15 @@ public async Task<CaseQueryResult> QueryCase(CaseQuery query)
             };
         }
 
+        public async Task<NotesQueryResult> QueryNotes(NotesQuery query)
```
Comment (ytqsl) [inline ess/src/API/EMBC.ESS/Resources/Cases/CaseRepository.cs:58]: can you roll this up into the main Query method entry point please?

File: ess/src/API/EMBC.ESS/Resources/Cases/Evacuations/Contract.cs [hunk 4]
```diff
@@ -133,6 +135,13 @@ public enum NoteType
         RecoveryPlan
     }
 
+    public enum NoteStatus
+    {
+        ReadOnly,
+        Editable,
```
Comment (ytqsl) [inline ess/src/API/EMBC.ESS/Resources/Cases/Evacuations/Contract.cs:141]: This is unlikely to be used anywhere, is it?

File: ess/src/API/EMBC.ESS/Resources/Cases/Evacuations/EvacuationRepository.cs [hunk 5]
```diff
@@ -337,6 +337,27 @@ public async Task<string> UpdateSecurityPhrase(string fileId, string securityPhr
             return evacuationFile.era_name;
         }
 
+        public async Task<IEnumerable<Note>> GetNotes(EvacuationFileNotesQuery query)
```
Comment (ytqsl) [inline ess/src/API/EMBC.ESS/Resources/Cases/Evacuations/EvacuationRepository.cs:340]: check the logic here, you may end up returning all notes in the system!

**Review by ytqsl — CHANGES_REQUESTED** [review]

File: ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs (hunk 1 above)
Comment (josephweinkamgov) [reply ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs:313]: The idea is that if note id is null then we are creating a new note and the file id is needed to link the note.
If note id is populated then the file id isn't needed.

So the only real bad condition is if they are both empty.

File: ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs (hunk 2 above)
Comment (josephweinkamgov) [reply ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs:320]: in my defense that name was already there =)

File: ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs (hunk 1 above)
Comment (ytqsl) [reply ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs:313]: wouldn't the file ID be mandatory and the not id will determine if this is update or insert? then you can always enforce the right file is linked to the note, and won't be changed in an update

File: ess/src/API/EMBC.ESS/Resources/Cases/Evacuations/Contract.cs [hunk 6]
```diff
@@ -113,13 +115,10 @@ public class Pet
         public string Quantity { get; set; }
     }
 
-    public class Note
+    public class Note : Case
```
Comment (ytqsl) [inline ess/src/API/EMBC.ESS/Resources/Cases/Evacuations/Contract.cs:118]: a note is not a case, it is part of the file. At the manager level, it is ok to query for a particular note or for the notes of a file, but at the repository level, querying only notes makes little sense

File: ess/src/API/EMBC.ESS.Shared.Contracts/Submissions/EvacuationFiles.cs [hunk 7]
```diff
@@ -137,8 +137,8 @@ public class Note
         public string Id { get; set; }
         public NoteType Type { get; set; }
         public string Content { get; set; }
-        public DateTime AddedOn { get; set; }
-        public DateTime ModifiedOn { get; set; }
+        public DateTime CreatedOn { get; set; }
```
Comment (ytqsl) [inline ess/src/API/EMBC.ESS.Shared.Contracts/Submissions/EvacuationFiles.cs:140]: are you trying to create conflicts for me? :P I'm working on this, if this is not required for this story, can you please undo these changes?

File: responders/src/API/EMBC.Responders.API/Controllers/RegistrationsController.cs [hunk 8]
```diff
@@ -247,31 +247,56 @@ public async Task<ActionResult<RegistrationResult>> CreateFileNote(string fileId
...
-        /// Updates a File Note
+        /// Updates a File Note's content
         /// </summary>
-        /// <param name="fileId">fileId</param>
         /// <param name="noteId">noteId</param>
         /// <param name="note">note</param>
-        /// <returns>newly created note id</returns>
-        [HttpPost("files/{fileId}/notes/{noteId}")]
+        /// <returns>updated note id</returns>
+        [HttpPost("notes/{noteId}/content")]
```
Comment (ytqsl) [inline responders/src/API/EMBC.Responders.API/Controllers/RegistrationsController.cs:255]: not needed to be that specific, also, you may not find the note and should return 404. Also, the files/{id} was a good thing, any reason to remove it?

File: responders/src/API/EMBC.Responders.API/Controllers/RegistrationsController.cs [hunk 9]
```diff
@@ -247,31 +247,56 @@ public async Task<ActionResult<RegistrationResult>> CreateFileNote(string fileId
...
+            return Ok(new EvacuationFileNotesResult { Id = id });
+        }
 
+        /// <summary>
+        /// Sets a File Note's isHidden field
+        /// </summary>
+        /// <param name="noteId">noteId</param>
+        /// <param name="note">note</param>
+        /// <returns>updated note id</returns>
+        [HttpPost("notes/{noteId}/hidden")]
```
Comment (ytqsl) [inline responders/src/API/EMBC.Responders.API/Controllers/RegistrationsController.cs:283]: notes should be nested under a specific file, they're not independent resources

**Review by ytqsl — CHANGES_REQUESTED** [review]
I aim to make your life harder today... :)

**Review by ytqsl — APPROVED** [review]
